contrib/gocql/gocql/observer.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package gocql
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/gocql/gocql"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// NewTracedSession returns a new session augmented with tracing.
+func NewTracedSession(cluster *gocql.ClusterConfig, opts ...WrapOption) (*gocql.Session, error) {
+	obs := NewObserver(cluster, opts...)
+	cfg := obs.cfg
+
+	if cfg.traceQuery {
+		cluster.QueryObserver = obs
+	}
+	if cfg.traceBatch {
+		cluster.BatchObserver = obs
+	}
+	if cfg.traceConnect {
+		cluster.ConnectObserver = obs
+	}
+	return cluster.CreateSession()
+}
+
+// TraceQuery can be used to enable tracing an individual *gocql.Query.
+func TraceQuery(q *gocql.Query, cluster *gocql.ClusterConfig, opts ...WrapOption) *gocql.Query {
+	obs := NewObserver(cluster, opts...)
+	return q.Observer(obs)
+}
+
+// TraceBatch can be used to enable tracing for an individual *gocql.Batch.
+func TraceBatch(b *gocql.Batch, cluster *gocql.ClusterConfig, opts ...WrapOption) *gocql.Batch {
+	obs := NewObserver(cluster, opts...)
```

Comment:
^ also one note, how to we pass the context here, for tracing? Because my understanding is that, as of today, we *need* to pass it somehow. Do we have to add a `.WithContext(ctx)` call or something?

Reply:
I think you should add that call, indeed. I'm thinking of removing `TraceQuery` and `TraceBatch` per your comment [here](https://github.com/DataDog/dd-trace-go/pull/2805#discussion_r1701976824). So, this won't longer apply, right?

Reply:
correct, in general the context is passed using `gocql.Query.WithContext` and `gocql.Batch.WithContext` functions. The Connect observer api does not provide the context.

But as @darccio says, since we are removing these 2 functions this no longer applies.